ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/CreateLoadReport/CreateLoadReport.cpp

```diff
@@ -233,21 +233,19 @@ void CreateLoadReport::runReport(const QStringList& selectedPhaseNames)
 
 void CreateLoadReport::setUtilityTraceOrconditionWithCodedValue(CodedValue codedValue)
 {
+  if (!m_baseCondition)
+    return;
+
   // Create a conditional expression with the CodedValue
   UtilityNetworkAttributeComparison* utilityNetworkAttributeComparison =
         new UtilityNetworkAttributeComparison(
           m_phasesCurrentAttribute,
           UtilityAttributeComparisonOperator::DoesNotIncludeAny,
           codedValue.code(), this);
 
-  UtilityTraceOrCondition* utilityTraceOrCondition = nullptr;
-
   // Chain it with the base condition using an OR operator.
-  if (utilityNetworkAttributeComparison && m_baseCondition)
```

Comment:
If `baseCondition` is false, `utilityTraceOrCondition` is not created and the function return without using `utilityNetworkAttributeComparison`. Test `baseCondition` in the beginning of the function to avoid the "leak".

Reply:
Again, there is no leak (even with quotes, "leak") since there is a parent. But good catch that we can rewrite the logic to avoid a heap allocation altogether in this case.